src/main/kotlin/com/example/tasks/taskRouter.kt

```diff
@@ -69,6 +70,17 @@ fun Route.tasks(context: DSLContext) {
                 }
             }
 
+            delete("/{id}") {
+                val userPrinciple = call.principal<UserPrinciple>()!!
+                if (userPrinciple.profile.role == UserRole.HUDDLE_AGENT) {
+                    val taskIDToBeDeleted = call.parameters["id"]
```

Comment:
```suggestion
                    val taskIDToBeDeleted = call.parameters["id"]!!
```
We need to make sure that the id is not null, a minor change otherwise looks great.